docs/spec/spec.md

```diff
@@ -0,0 +1,234 @@
+# 
+# + dN - The distinguished name of the entry
+# + entry - The information to add
+# + return - A `ldap:Error` if the operation fails or `ldap:LdapResponse` if successfully created
+remote isolated function add(string dN, Entry entry) returns LdapResponse|Error;
+```
+
+#### 3.1.1 DNs and RDNs
+
+The distinguished name (`DN`) of an entry is used to uniquely identify the entry and its location within the directory information tree (`DIT`) hierarchy. It's similar to how a file path specifies the location of a file in a filesystem.
+
+A `DN` consists of one or more comma-separated components known as relative distinguished names (`RDN`s). Typically, the leftmost component in the `DN` is considered the `RDN` for that entry. [Learn more](https://ldap.com/ldap-dns-and-rdns/)
+
+### 3.2 Modify operation
+
+Updates information of an entry in a directory server.
+
+```ballerina
+# Updates information of an entry.
+#
+# + dN - The distinguished name of the entry
+# + entry - The information to update
+# + return - A `ldap:Error` if the operation fails or `LdapResponse` if successfully updated
+remote isolated function modify(string dN, Entry entry) returns LdapResponse|Error;
+```
+
+### 3.3 ModifyDn operation
+
+Renames an entry in a directory server.
+
+```ballerina
+# Renames an entry in a directory server.
+#
+# + currentDn - The current distinguished name of the entry
+# + newRdn - The new relative distinguished name
+# + deleteOldRdn - A boolean value to determine whether to delete the old RDN
+# + return - A `ldap:Error` if the operation fails or `ldap:LdapResponse` if successfully renamed
+remote isolated function modifyDn(string currentDn, string newRdn, boolean deleteOldRdn = false) returns LdapResponse|Error;
+```
+
+### 3.4 Compare operation
+
+Determines whether a given entry has a specified attribute value.
+
+```ballerina
+# Determines whether a given entry has a specified attribute value.
+# 
+# + dN - The distinguished name of the entry
+# + attributeName - The name of the target attribute for which the comparison is to be performed
+# + assertionValue - The assertion value to verify within the entry
+# + return - A `boolean` value indicating whether the values match, or an `ldap:Error` if the operation fails
+remote isolated function compare(string dN, string attributeName, string assertionValue) returns boolean|Error;
+```
+
+### 3.5 Search operation
+
+Returns a record containing search result entries and references that match the given search parameters.
+
+```ballerina
+# Returns a record containing search result entries and references that match the given search parameters.
+#
+# + baseDn - The base distinguished name of the entry
+# + filter - The filter to be used in the search
+# + scope - The scope of the search
+# + return - An `ldap:SearchResult` if successful, or else `ldap:Error`
+remote isolated function search(string baseDn, string filter, SearchScope scope) returns SearchResult|Error;
```

Comment:
Shall we add a section to describe `ldap:SearchScope` as well ?

Reply:
Fixed with https://github.com/ballerina-platform/module-ballerina-ldap/pull/58/commits/d393c7dc09629ff0a3b5de1ba92727c9c9538dc8